src/core/dbus-execute.c

```diff
@@ -1775,20 +2146,43 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
-        } else if (streq(name, "SELinuxContext")) {
+        } else if (STR_IN_SET(name, "SELinuxContext", "AppArmorProfile", "SmackProcessLabel")) {
+                bool ignore;
                 const char *s;
 
-                r = sd_bus_message_read(message, "s", &s);
+                r = sd_bus_message_enter_container(message, 'r', "bs");
```

Comment:
you are breaking the D-Bus API here, as SELinuxContext so far expected a string, not a `(bs)`... I'd leave SELinuxContext the way it is, and only add AppArmorProfile and SmackProcessLabel for now, in the new style